modopt/opt/algorithms/admm.py

```diff
@@ -0,0 +1,343 @@
+"""ADMM Algorithms."""
+import numpy as np
+
+from modopt.base.backend import get_array_module
+from modopt.opt.algorithms.base import SetUp
+from modopt.opt.cost import AbstractcostObj
+
+
+class ADMMcostObj(AbstractcostObj):
+    r"""Cost Object for the ADMM problem class.
+
+    Compute :math:`f(u)+g(v) + \tau \| Au +Bv - b\|^2`
+
+    Parameters
+    ----------
+    cost_funcs: 2-tuples of callable
+        f and g function.
+    A : OperatorBase
+        First Operator
+    B : OperatorBase
+        Second Operator
```

Comment:
As I mentioned in the previous PR, I would really like to minimise the use of single character variables where possible.

Reply:
In this context the variable name are chosen to stick to the paper implementation, and this also the case for other algorithms in ModOpt (POGM for instance has u,x,y,z). As we are writing scientific code, I think its best to stick to the scientific notation, easing the matching between code and the algorithm descriptions in the paper. Also, I don't find any other name to use instead.